greenpeace/planet4-plugin-gutenberg-blocks
# PLANET-5978: Theme creator experiment

REVIEW INSTRUCTIONS:
* The most important thing to look out for in this PR is that it should not affect the site unless the feature is turned on.
* So the code in `/theme` doesn't need to be extensively reviewed, we just need to be sure it has no effect whatsoever if the feature is off.

This is currently deployed on my dev instance: https://www-dev.greenpeace.org/pieterdev 
Alt + click on an element to open the editor for it (works on anything, as long as there is something that can be configured)

A while back I started trying to make a theme editor based entirely on information extracted from the selectors CSS variables are in (and a few assumptions about the naming of the variables). Then matching the clicked element against those to get all available settings for it.

I worked on it from time to time, now I feel it's at a point where it could be useful for designers on the project, since it allows for instant prototyping of any property. In fact it's not even a prototype, it's the real thing. Since we're currently in the process of setting up a design system, the ability to instantly prototype things can probably help a lot there.

~~I also included the styleguide with a bunch of added variables.~~ Removed again for merging. Unfortunately that also removes most of the themable things :sad_panda:

So after this change, what work is needed to set up a UI control for a given CSS value?
1. Add a variable around the value
2. If npm watch is on, you just need to refresh so that it can fetch the new config file that was built

~~TODO blocker for merging:~~
* ~~Integrate in a way that it can be used by logged in users if beta feature is turned on without changing anything for non logged in.~~ Done: master theme will have a toggle to enable the editor, and it's loaded with separate files.

TODO:
* Consistent handling of variables for the same property but different screen sizes
* Cleaner webpack implementation, or maybe even parse the CSS files in the browser?
* Custom controls for more properties
* Combining :active,:focus and others into the same control as the regular version of the a property
* Apply consistent variable naming.
* Consistent way to handle complex properties (e.g. either add a control for `margin` or always split up into single value properties `margin-left`,`margin-bottom`,...)
